src/field.cpp

```diff
@@ -1635,32 +1635,39 @@ void map::player_in_field( player &u )
         } break;
 
         case fd_acid:
-            //TODO: Add resistance to this with rubber shoes or something?
+        {
             // Assume vehicles block acid damage entirely,
             // you're certainly not standing in it.
-            if (veh) {
+            if( veh != nullptr ) {
                 break;
             }
-            if (cur->getFieldDensity() > 0 && !u.has_trait("ACIDPROOF")) {
+            if( cur->getFieldDensity() > 0 && !u.has_trait("ACIDPROOF") ) {
                 u.add_msg_player_or_npc(m_bad, _("The acid burns your legs and feet!"), _("The acid burns <npcname>s legs and feet!"));
```

Comment:
It looks like there's a chance for the player to take no damage due to environmental resistance, but this message will still print.